docs/en/stack/ml/nlp/ml-nlp-elastic-rerank.asciidoc

```diff
@@ -0,0 +1,242 @@
+
+[discrete]
+[[ml-nlp-rerank-availability]]
+== Availability and requirements 
+
+IMPORTANT: This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.
+
+[discrete]
+[[ml-nlp-rerank-availability-serverless-]]
+=== Elastic Cloud Serverless
+
+Elastic Rerank is available in Elasticsearch Serverless projects as of November 25, 2024.
+
+[discrete]
+[[ml-nlp-rerank-availability-elastic-stack]]
+=== Elastic Cloud Hosted and self-managed deployments
+
+Elastic Rerank is available in Elastic Stack version 8.17+:
+
+* To use Elastic Rerank, you must have the appropriate subscription level or the trial period activated.
+* A 4GB ML node
++
+[IMPORTANT]
+====
+Deploying the Elastic Rerank model in combination with ELSER (or other hosted models) requires at minimum an 8GB ML node. Please note that the current maximum size for trial ML nodes is 4GB (defaults to 1GB).
+====
+
+[discrete]
+[[ml-nlp-rerank-deploy]]
+== Download and deploy
+
+To download and deploy Elastic Rerank, use the {ref}/infer-service-elasticsearch.html[create inference API] to create an Elasticsearch service `rerank` endpoint.
+
+[discrete]
+[[ml-nlp-rerank-deploy-steps]]
+=== Create an inference endpoint
+
+. In {kib}, navigate to the *Dev Console*.
+
+. Create an {infer} endpoint with the Elastic Rerank service by running:
++
+[source,console]
+----------------------------------
+PUT _inference/text_similarity/my-rerank-model
+{
+  "service": "elasticsearch",
+  "service_settings": {
+    "adaptive_allocations": {
+      "enabled": true,
+      "min_number_of_allocations": 1,
+      "max_number_of_allocations": 10
+    },
+    "num_threads": 1,
+    "model_id": ".elastic-rerank"
+  }
+}
+----------------------------------
++
+NOTE: The API request automatically downloads and deploys the model. This example uses <<ml-nlp-auto-scale,autoscaling>> through adaptive allocation.
+
+After creating the Elastic Rerank {infer} endpoint, it's ready to use with a {ref}/retriever.html#text-similarity-reranker-retriever-example-elastic-rerank[`text_similarity_reranker`] retriever.
+
+[discrete]
+[[ml-nlp-rerank-limitations]]
+== Limitations
+
+* English language only
+* Maximum context window of 512 tokens
++
+When using the {ref}/semantic-text.html[`semantic_text` field type], text is divided into chunks. By default, each chunk contains 250 words (approximately 400 tokens). Be cautious when increasing the chunk size - if the combined length of your query and chunk text exceeds 512 tokens, the model won't have access to the full content.
++
+When the combined inputs exceed the 512 token limit, a "balanced" truncation strategy is used. If both the query and input text are longer than 255 tokens each then both are truncated, otherwise the longest is truncated.
```

Comment:
Could we just remove the quotation marks? I think it could be a source of confusion.
```suggestion
When the combined inputs exceed the 512 token limit, a balanced truncation strategy is used. If both the query and input text are longer than 255 tokens each then both are truncated, otherwise the longest is truncated.
```